src/compiler/checker.ts

```diff
@@ -1636,9 +1636,9 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
         getResolvedSignature: (node, candidatesOutArray, argumentCount) =>
             getResolvedSignatureWorker(node, candidatesOutArray, argumentCount, CheckMode.Normal),
         getResolvedSignatureForStringLiteralCompletions: (call, editingArgument, candidatesOutArray) =>

```

Comment:
TODO: I should probably also figure out a test case for string literal completions

Reply:
it actually works just OK with string completions because within that call the `editingArgument` is always there and thus the call to `getResolvedSignature` is already wrapped with `runWithInferenceBlockedFromSourceNode` and that already includes the logic for blocking the resolved signature caching

nevertheless... I'm gonna add an extra regression test case for this in a moment even though it's not strictly needed right now